test-suite/js/src/test/scala/org/scalajs/testsuite/niobuffer/ByteBufferJSFactories.scala

```diff
@@ -0,0 +1,27 @@
+package org.scalajs.testsuite.niobuffer
+
+import scala.scalajs.js.JSConverters._
+import scala.scalajs.js.typedarray._
+
+import java.nio._
+
+object ByteBufferJSFactories {
+  import BufferFactory._
+
+  class AllocDirectByteBufferFactory extends ByteBufferFactory {
+    def allocBuffer(capacity: Int): ByteBuffer =
+      ByteBuffer.allocateDirect(capacity)
+  }
```

Comment:
Oh, now I understand, it is duplicated and I was looking at the one in `shared`. I will remove it from here.